RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromTTracksEDProducer.cc

```diff
@@ -0,0 +1,169 @@
+
+  edm::ESHandle<Propagator> propagatorOppositeH;
+  es.get<TrackingComponentsRecord>().get(thePropagatorName, propagatorOppositeH);
+  std::unique_ptr<Propagator> propagatorOpposite = SetPropagationDirection(*propagatorOppositeH, oppositeToMomentum);
+
+  // Get vector of Detector layers
+  edm::Handle<MeasurementTrackerEvent> measurementTrackerH;
+  ev.getByToken(theMeasurementTrackerTag, measurementTrackerH);
+  std::vector<BarrelDetLayer const*> const& tob = measurementTrackerH->geometricSearchTracker()->tobLayers();
+  std::vector<ForwardDetLayer const*> const& tecPositive =
+      tmpTkGeometryH->isThere(GeomDetEnumerators::P2OTEC)
+          ? measurementTrackerH->geometricSearchTracker()->posTidLayers()
+          : measurementTrackerH->geometricSearchTracker()->posTecLayers();
+  std::vector<ForwardDetLayer const*> const& tecNegative =
+      tmpTkGeometryH->isThere(GeomDetEnumerators::P2OTEC)
+          ? measurementTrackerH->geometricSearchTracker()->negTidLayers()
+          : measurementTrackerH->geometricSearchTracker()->negTecLayers();
+
+  /// Surface used to make a TSOS at the PCA to the beamline
+  Plane::PlanePointer dummyPlane = Plane::build(Plane::PositionType(), Plane::RotationType());
+
+  // Loop over the L1's and make seeds for all of them:
+  std::vector<TTTrack<Ref_Phase2TrackerDigi_>>::const_iterator it;
+  for (it = trks->begin(); it != trks->end(); it++) {
+    const TTTrack<Ref_Phase2TrackerDigi_>& l1 = (*it);
+
+    std::unique_ptr<std::vector<TrajectorySeed>> out(new std::vector<TrajectorySeed>());
+    FreeTrajectoryState fts = trajectoryStateTransform::initialFreeStateTTrack(l1, magfieldH.product(), false);
+    dummyPlane->move(fts.position() - dummyPlane->position());
+    TrajectoryStateOnSurface tsosAtIP = TrajectoryStateOnSurface(fts, *dummyPlane);
+
+    unsigned int numSeedsMade = 0;
+    //BARREL
+    if (std::abs(l1.momentum().eta()) < theMaxEtaForTOB) {
+      for (auto it = tob.rbegin(); it != tob.rend(); ++it) {  //This goes from outermost to innermost layer
+        findSeedsOnLayer(**it, tsosAtIP, *(propagatorAlong.get()), l1, estimatorH, numSeedsMade, out);
+      }
+    }
+    if (std::abs(l1.momentum().eta()) > theMinEtaForTEC && std::abs(l1.momentum().eta()) < theMaxEtaForTOB) {
+      numSeedsMade = 0;  // reset num of seeds
+    }
+    //ENDCAP+
+    if (l1.momentum().eta() > theMinEtaForTEC) {
+      for (auto it = tecPositive.rbegin(); it != tecPositive.rend(); ++it) {
+        findSeedsOnLayer(**it, tsosAtIP, *(propagatorAlong.get()), l1, estimatorH, numSeedsMade, out);
+      }
+    }
+    //ENDCAP-
+    if (l1.momentum().eta() < -theMinEtaForTEC) {
+      for (auto it = tecNegative.rbegin(); it != tecNegative.rend(); ++it) {
+        findSeedsOnLayer(**it, tsosAtIP, *(propagatorAlong.get()), l1, estimatorH, numSeedsMade, out);
+      }
+    }
+    for (std::vector<TrajectorySeed>::iterator it = out->begin(); it != out->end(); ++it) {
+      result->push_back(*it);
+    }
```

Comment:
```suggestion
    std::copy(out->begin(), out->end(), std::back_inserter(result));
```

Reply:
This doesn't seem to work. It gives me the following error and I can't seem to solve it:
```/cvmfs/cms.cern.ch/slc7_amd64_gcc820/external/gcc/8.2.0-bcolbf/include/c++/8.4.0/bits/stl_iterator.h:490:7: error: no type named 'value_type' in 'class std::unique_ptr<std::vector<TrajectorySeed> >'
       operator=(const typename _Container::value_type& __value```. Do you have any suggestions?